cli/pcluster/config/param_types.py

```diff
@@ -519,6 +519,38 @@ def to_cfn(self):
         return cfn_params
 
 
+class ComputeInstanceTypeParam(Param):
+    """
+    Class to manage the compute_instance_type configuration parameter.
+
+    We need this class since a different default value should be used
+    for the CFN "ComputeInstanceType" parameter when the scheduler is awsbatch.
+    """
+
+    @handle_no_section_error
+    def from_file(self, config_parser):
+        """
+        Use different default value when the scheduler is awsbatch.
+
+        If the scheduler is awsbatch and the config file does not specify a
+        value for this parameter, use "optimal" instead of the usual default.
+        """
+        section_name = _get_file_section_name(self.section_key, self.section_label)
+        if config_parser.has_option(section_name, self.key):
+            self.value = config_parser.get(section_name, self.key).strip()
+        elif (
+            config_parser.has_option(section_name, "scheduler")
```

Comment:
Use `get_param_value()` which handles the option not-existing scenario. So this becomes just:

```python
elif self.get_param_value("scheduler") == "awsbatch":
      self.value = "optimal"
```

Reply:
The reason I opted for this is because when the `compute_instance_type` parameter is parsed before the `scheduler` parameter, the value returned by `get_param_value` would be the default value. Sean suggested adding a method to the `Param` class for this type of check, given that other default values for parameters within the same section is a potentially common scenario.